database/postgres/postgres.go

```diff
@@ -340,12 +340,28 @@ func (p *Postgres) ensureVersionTable() (err error) {
 	defer func() {
 		if e := p.Unlock(); err == nil {
 			err = e
+		} else if e != nil {
+			err = &database.Error{OrigErr: err, Err: e.Error()}
```

Comment:
Don't use `database.Error` since it's meant to wrap database errors.
In this scenario, the appropriate usage of `database.Error` would be: `database.Error{OrigErr: e, Err: "Failed to unlock pg advisory lock while ensuring schema version table"}`, which should still reference any original error.